core/src/test/java/kafka/testkit/KafkaClusterTestKit.java

```diff
@@ -651,4 +648,12 @@ private void waitForAllFutures(List<Entry<String, Future<?>>> futureEntries)
             log.debug("{} successfully shut down.", entry.getKey());
         }
     }
+
+    private void waitForAllThreads() throws InterruptedException {
+        Set<Long> threadIds = threadFactory.getThreadIds();
+        // make sure all threads from threadFactory is shut down
+        TestUtils.waitForCondition(() -> Thread.getAllStackTraces().keySet()
```

Comment:
we should wait it until those threads disappear from `getAllStackTraces`, right?